### PR TITLE
fix: added validator, password and email logo fixes

### PR DESCRIPTION
**Pull request by masterchief164**
This PR fixes the upper and lower case validation, password set and email logo issues. This also adds support for the new card authentication flow.

### CODE REVIEW

File: packages/cysync-core/src/context/lockscreen.tsx [hunk 1]
```diff
@@ -84,6 +84,7 @@ export const LockscreenProvider: React.FC<LockscreenProviderProps> = ({
     if (password) {
       const encryptionKey = await passwordUtils.getEncryptionKey(password);
       await db.changeEncryptionKey(encryptionKey);
+      setIsPasswordSet(true);
       logger.info('Password successfully changed');
     } else {
       await db.changeEncryptionKey(undefined);
```
Comment (irshadCypherock) [inline packages/cysync-core/src/context/lockscreen.tsx:90]:
```suggestion
      await db.changeEncryptionKey(undefined);
      setIsPasswordSet(false);
```

**Review by irshadCypherock — CHANGES_REQUESTED** [review]

File: packages/cysync-core/src/context/lockscreen.tsx (hunk 1 above)
Comment (masterchief164) [reply packages/cysync-core/src/context/lockscreen.tsx:90]: Why are we setting this false? We have added a password, right?

File: packages/cysync-core/src/constants/i18n/en.ts [hunk 2]
```diff
@@ -166,11 +166,16 @@ const en = {
...
       subtext:
         'Do not disconnect your device while the operation is being done',
-      success: 'Your X1 Vault is successfully authenticated',
+      success: {
+        title: 'Your X1 Vault is successfully authenticated',
+        subtext: 'Wait while we take you to the next screen',
+      },
       error: 'X1 Vault authentication has failed',
+      errorSubtext:
+        'Device seems to be Compromised. Contact Cypherock support immediately',
```
Comment (irshadCypherock) [inline packages/cysync-core/src/constants/i18n/en.ts]: Unused variable. It has need moved to `deviceErrors`. Search for `DEVICE_AUTH_FAILED`

File: packages/ui/src/components/atoms/Input/PasswordInput.tsx [hunk 3]
```diff
@@ -20,7 +20,7 @@ export const PasswordInput: FC<PasswordInputProps> = props => {
     <Input
       type={showPassword ? 'text' : 'password'}
       {...props}
-      postfixIcon={showPassword ? visibilityHideIcon : visibilityIcon}
+      postfixIcon={!showPassword ? visibilityHideIcon : visibilityIcon}
```
Comment (irshadCypherock) [inline packages/ui/src/components/atoms/Input/PasswordInput.tsx]:
```suggestion
      postfixIcon={showPassword ? visibilityIcon : visibilityHideIcon}
```

File: packages/cysync-core/src/pages/OnBoarding/JoystickTraining/Dialogs/index.tsx [hunk 4]
```diff
@@ -20,15 +21,23 @@ export const JoystickTrainingDialog: React.FC = () => {
...
+        setState(s);
+        await sleep(600);
+        setState(TrainJoystickStatus.TRAIN_JOYSTICK_LEFT + 1);
+      }
     });
-    setState(TrainJoystickStatus.TRAIN_JOYSTICK_CENTER);
+    setState(TrainJoystickStatus.TRAIN_JOYSTICK_CENTER + 1);
+    await sleep(600);
+    setState(TrainJoystickStatus.TRAIN_JOYSTICK_CENTER + 2);
+    navigateTo(routes.onboarding.cardTraining.path, 3000);
```
Comment (irshadCypherock) [inline packages/cysync-core/src/pages/OnBoarding/JoystickTraining/Dialogs/index.tsx]: Use the navigation path defined on the line `51`. Also remove the `useEffect`, if we are redirecting directly after successful run.

**Review by irshadCypherock — CHANGES_REQUESTED** [review]

**Comment by masterchief164** [comment]
The e2e tests will fail because we are expecting to find a full screen window of the usage page but due to the new media queries the mini screen comes up.

File: packages/ui/src/themes/screens.styled.ts [hunk 5]
```diff
@@ -3,6 +3,8 @@ export const screens = {
   md: '1024px',
   lg: '1440px',
   xl: '2560px',
+
+  hlg: '950px',
```
Comment (irshadCypherock) [inline packages/ui/src/themes/screens.styled.ts]: What does `hlg` stand for? Looks like it needs to be `sm`

File: packages/ui/src/components/atoms/LangDisplay.tsx [hunk 6]
```diff
@@ -5,13 +5,26 @@ interface LangDisplayProps {
...
+      </React.Fragment>
+    ))}
+  </>
 );
+const BaseLangDisplay: React.FC<LangDisplayProps> = ({ text, variables }) => {
+  const parsedText = parseVariables(text, variables);
+  const parsedTextWithLineBreaks = parseLineBreaks(parsedText);
+  // eslint-disable-next-line react/jsx-no-useless-fragment
+  return <>{parsedTextWithLineBreaks}</>;
+};
```
Comment (irshadCypherock) [inline packages/ui/src/components/atoms/LangDisplay.tsx]: Have you tried something like this?

```suggestion
const BaseLangDisplay: React.FC<LangDisplayProps> = ({ text, variables }) => (
  // eslint-disable-next-line react/jsx-no-useless-fragment
  <span style={{ whiteSpace: 'pre-wrap' }}>{getText(text, variables)}</span>
);
```

**Review by irshadCypherock — CHANGES_REQUESTED** [review]

File: packages/ui/src/components/atoms/LangDisplay.tsx (hunk 6 above)
Comment (masterchief164) [reply packages/ui/src/components/atoms/LangDisplay.tsx:27]: This didn't work.

File: packages/ui/src/themes/screens.styled.ts (hunk 5 above)
Comment (masterchief164) [reply packages/ui/src/themes/screens.styled.ts]: hlg means height large and we can't rename it to sm as it's large for heights

File: packages/ui/src/components/atoms/LangDisplay.tsx (hunk 6 above)
Comment (masterchief164) [reply packages/ui/src/components/atoms/LangDisplay.tsx]: This breaks the tests as we are unable to get items by roles as adding a span adds even the headings are recognized as headings. Moreover, we shouldn't add another element where we are adding a text.

**Review by irshadCypherock — APPROVED** [review]

**Pull request merged**